python/huggingfaceserver/huggingfaceserver/test_model.py

```diff
@@ -175,5 +175,28 @@ def test_input_truncation():
     assert response == {"predictions": [1]}
 
 
+def test_input_padding_with_pad_token_not_specified():
+    model = HuggingfaceModel(
+        "openai-gpt",
+        {
+            "model_id": "openai-community/openai-gpt",
+            "task": MLTask.text_generation.value,
+            "disable_special_tokens": True,
+        },
+    )
+    model.load()
+
+    # inputs with different lengths will throw an error
+    # unless padding token is configured.
+    # openai-gpt model does not specify the pad token, so the fallback pad token should be added.
+    assert model.tokenizer.pad_token == "[PAD]"
+    assert model.tokenizer.pad_token_id is not None
+    request_one = "my name is merve and my favorite color is blue."
+    request_two = "my name is teven and i am"
+    response = asyncio.run(model({"instances": [request_one, request_two]}, headers={}))
+    assert request_one in response["predictions"][0]
+    assert request_two in response["predictions"][1]
```

Comment:
I am asserting it this way because the output changes occasionally.

Reply:
I think we can specify the temperature to 0 to get deterministic response